apps/cowswap-frontend/src/modules/tenderly/hooks/useTenderlyBundleSimulation.ts

```diff
@@ -0,0 +1,69 @@
+import { useCallback } from 'react'
+
+import { useWalletInfo } from '@cowprotocol/wallet'
+
+import { bundleSimulation } from '../utils/bundleSimulation'
+import { useTopTokenHolders } from './useTopTokenHolders'
+import { getTokenTransferInfo } from '../utils/getTokenTransferInfo'
+import { useOrderParams } from 'modules/hooksStore/hooks/useOrderParams'
+import { useTokenContract } from 'common/hooks/useContract'
+import useSWR from 'swr'
+import { useHooks } from 'modules/hooksStore'
+import { generateNewSimulationData, generateSimulationDataToError } from '../utils/generateSimulationData'
+
+export function useTenderlyBundleSimulateSWR() {
+  const { account, chainId } = useWalletInfo()
+  const { preHooks, postHooks } = useHooks()
+  const orderParams = useOrderParams()
+  const tokenSell = useTokenContract(orderParams?.sellTokenAddress)
+  const tokenBuy = useTokenContract(orderParams?.buyTokenAddress)
+  const buyAmount = orderParams?.buyAmount
+
+  const { data: buyTokenTopHolders, isValidating: isTopTokenHoldersValidating } = useTopTokenHolders({
+    tokenAddress: tokenBuy?.address,
+    chainId,
+  })
```

Comment:
We just use it when there are post hooks. However, I thought that running two requests in sequence could be a little slow for the user (find the holders and run the simulation). Also, the bff endpoint has some caches, so I think that this will not be so expensive.

Reply:
> We just use it when there are post hooks.

Even if this is used only when there are post hooks, it'll be called on every simulation even if not needed, correct?
And sure, it'll be cached and not hit upstream again unless the buy token changes, but it will still be called needlessly when there are no post hooks, no?
We could for example trigger it when the user opens the post-hook dialog, if the issue is the speed.

Reply:
Makes sense! refactored the useSwr to a jotai atom with storage and timed cache. With that we will only do this request when we need it.